assets/js/app.js

```diff
@@ -63,32 +63,16 @@ window.stencilBootstrap = function stencilBootstrap(pageType, contextJSON = null
     return {
         load() {
             $(async () => {
-                let globalClass;
-                let pageClass;
-                let PageClass;
-
-                // Finds the appropriate class from the pageType.
-                const pageClassImporter = pageClasses[pageType];
-                if (typeof pageClassImporter === 'function') {
-                    PageClass = (await pageClassImporter()).default;
-                }
-
+                // Load globals
                 if (loadGlobal) {
-                    globalClass = new Global();
-                    globalClass.context = context;
+                    Global.load(context);
                 }
 
-                if (PageClass) {
-                    pageClass = new PageClass(context);
-                    pageClass.context = context;
-                }
-
-                if (globalClass) {
-                    globalClass.load();
-                }
-
-                if (pageClass) {
-                    pageClass.load();
+                // Find the appropriate page loader based on pageType
+                const pageClassImporter = pageClasses[pageType];
+                if (typeof pageClassImporter === 'function') {
+                    const PageClass = (await pageClassImporter()).default;
```

Comment:
Yeah, `await` is non-blocking in JS, so not like the `Await` object in Scala. Fun fact, there's an async/await macro library for Scala that does the same thing: https://github.com/scala/scala-async 

It basically rewrites the code to use `.map` and `.flatMap` in the compiler.

Reply:
@icatalina Yes, we are running everything through babel in stencil CLI.

Reply:
await is diff than a promise because it blocks

Reply:
const pageClass = await xx  <-- how can this be async what if somone uses the pageClass val before the await is resolved?

Reply:
actually....i think this might be causing the test failure RegisteredCustomerMessagesUITest::testRequiredFieldWhileCreatingMessage

Reply:
@tpietsch for the `const pageClass = await xx` example, it's simply sugar for using a promise.then, so it will not set the pageClass val until after the await is resolved.  but it's non-blocking in that the thread is released when invoking await to handle other things.

Reply:
it just kinda not my face cause now you have a variable in scope that you can use where the value  might change right? first its undef then its def and between that time it can be read from incorrectly
In a normal Promise(resolve -> resolve.data) <-- you are more explicit about when the data is being used here it seems like it could cause intermittent weirdness. And also the variable is declared as a const which i would expect as non-changing value